src/connector/src/parser/plain_parser.rs

```diff
@@ -393,4 +451,69 @@ mod tests {
+            Arc::new(source_ctx),
+        )
+        .await
+        .unwrap();
+        let mut builder = SourceStreamChunkBuilder::with_capacity(columns, 0);
+
+        let msg = r#"{"schema":null,"payload": { "databaseName": "mydb", "ddl": "ALTER TABLE test add column v2 varchar(32)", "schemaName": null, "source": { "connector": "mysql", "db": "mydb", "file": "binlog.000065", "gtid": null, "name": "RW_CDC_1", "pos": 234, "query": null, "row": 0, "sequence": null, "server_id": 1, "snapshot": "false", "table": "test", "thread": null, "ts_ms": 1718354727000, "version": "2.4.2.Final" }, "tableChanges": [ { "id": "\"mydb\".\"test\"", "table": { "columns": [ { "autoIncremented": false, "charsetName": null, "comment": null, "defaultValueExpression": null, "enumValues": null, "generated": false, "jdbcType": 4, "length": null, "name": "id", "nativeType": null, "optional": false, "position": 1, "scale": null, "typeExpression": "INT", "typeName": "INT" }, { "autoIncremented": false, "charsetName": null, "comment": null, "defaultValueExpression": null, "enumValues": null, "generated": false, "jdbcType": 2014, "length": null, "name": "v1", "nativeType": null, "optional": true, "position": 2, "scale": null, "typeExpression": "TIMESTAMP", "typeName": "TIMESTAMP" }, { "autoIncremented": false, "charsetName": "utf8mb4", "comment": null, "defaultValueExpression": null, "enumValues": null, "generated": false, "jdbcType": 12, "length": 32, "name": "v2", "nativeType": null, "optional": true, "position": 3, "scale": null, "typeExpression": "VARCHAR", "typeName": "VARCHAR" } ], "comment": null, "defaultCharsetName": "utf8mb4", "primaryKeyColumnNames": [ "id" ] }, "type": "ALTER" } ], "ts_ms": 1718354727594 }}"#;
+        let cdc_meta = SourceMeta::DebeziumCdc(DebeziumCdcMeta::new(
+            "mydb.test".to_string(),
+            0,
+            cdc_message::CdcMessageType::SchemaChange,
+        ));
+        let msg_meta = MessageMeta {
+            meta: &cdc_meta,
+            split_id: "1001",
+            offset: "",
+        };
+
+        let res = parser
+            .parse_one_with_txn(
+                None,
+                Some(msg.as_bytes().to_vec()),
+                builder.row_writer().with_meta(msg_meta),
+            )
+            .await;
+
+        let res = res.unwrap();
+        match res {
+            ParseResult::SchemaChange(schema_change) => {
+                assert_eq!(schema_change.table_changes.len(), 1);
+                let table_change = &schema_change.table_changes[0];
+                assert_eq!(table_change.cdc_table_name, "mydb.test");
+                assert_eq!(table_change.change_type, TableChangeType::Alter);
+                assert_eq!(table_change.columns.len(), 3);
+                let column_names = table_change.columns.iter().map(|c| c.name()).collect_vec();
+                assert_eq!(column_names, vec!["id", "v1", "v2"]);
+            }
+            _ => panic!("unexpected parse result: {:?}", res),
+        }
```

Comment:
Just a hint in case you don't know: You can use [expect_test](https://docs.rs/expect-test/latest/expect_test/). It can auto update the actual output string, so you don't need to manually `assert_eq` https://github.com/rust-analyzer/expect-test#expect-test
```suggestion
        expect_test::expect![  ].assert_debug_eq(&res);
```